firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -789,13 +789,32 @@ extension BrowserViewController: WKNavigationDelegate {
         }
 
         if let url = error.userInfo[NSURLErrorFailingURLErrorKey] as? URL {
+            guard var errorURLpath = URLComponents(string: "\(InternalURL.baseUrl)/\(ErrorPageHandler.path)" ) else { return }
```

Comment:
There are two places in this code where we `return` from a guard... but I wonder if we want to do that?

It looks like if `isNativeErrorPageEnabled` and `isNICErrorPageEnabled` are both false, we fall back to this line:
```
ErrorPageHelper(certStore: profile.certStore).loadPage(error, forUrl: url, inWebView: webView)
```

So if we can't get a `errorURLpath` or `errorPageURL`, should we also fallback to the `ErrorPageHelper`? 🤔 

(I might just be missing some context for your error page work)

Reply:
if isNativeErrorPageEnabled = false and isNICErrorPageEnabled = false, we should show the old error page, hence we fall back to ErrorPageHelper.

Reply:
Ok that makes sense!

So what happens if either `URLComponents(string: "\(InternalURL.baseUrl)/\(ErrorPageHandler.path)"` or `errorURLpath.url` return nil? 

It looks like the `guard` `else` just does a `return` in those 2 cases. Should we somehow return `ErrorPageHelper` instead? (Might need a little restructuring)

I do realize that these guard returns should never be called. 😅  So it's not a very important edge case. If you think it makes sense to leave the code like this, I'm fine with that. 👍

Reply:
Oh! Got your point. I will restructure the logic.